File: airbyte-cdk/java/airbyte-cdk/core/src/testFixtures/java/io/airbyte/cdk/testutils/NonContainer.java
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.cdk.testutils;

import org.testcontainers.containers.JdbcDatabaseContainer;

/**
 * This is used when a source (such as Snowflake) relies on an always-on resource and therefore
 * doesn't need an actual container. compatible
 */
public class NonContainer extends JdbcDatabaseContainer<NonContainer> {

  private final String username;
  private final String password;
  private final String jdbcUrl;

  private final String driverClassName;

  public NonContainer(final String userName,
                      final String password,
                      final String jdbcUrl,
                      final String driverClassName,
                      final String dockerImageName) {
    super(dockerImageName);
    this.username = userName;
    this.password = password;
    this.jdbcUrl = jdbcUrl;
    this.driverClassName = driverClassName;
  }

  @Override
  public String getDriverClassName() {
    return driverClassName;
  }

  @Override
  public String getJdbcUrl() {
    return jdbcUrl;
  }

  @Override
  public String getUsername() {
    return username;
  }

  @Override
  public String getPassword() {
    return password;
  }

  @Override
  protected String getTestQueryString() {
    return "SELECT 1";
  }

}

File: airbyte-cdk/java/airbyte-cdk/core/src/testFixtures/kotlin/io/airbyte/cdk/testutils/NonContainer.kt
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */
package io.airbyte.cdk.testutils

import org.testcontainers.containers.JdbcDatabaseContainer

/**
 * This is used when a source (such as Snowflake) relies on an always-on resource and therefore
 * doesn't need an actual container. compatible
 */
class NonContainer(
    private val username: String?,
    private val password: String?,
    private val jdbcUrl: String?,
    private val driverClassName: String?,
    dockerImageName: String
) : JdbcDatabaseContainer<NonContainer>(dockerImageName) {
    override fun getDriverClassName(): String? {
        return driverClassName
    }

    override fun getJdbcUrl(): String? {
        return jdbcUrl
    }

    override fun getUsername(): String? {
        return username
    }

    override fun getPassword(): String? {
        return password
    }

    override fun getTestQueryString(): String? {
        return "SELECT 1"
    }
}
